docs/docs/how-to/setup-intant-with-gocli.mdx

```diff
@@ -0,0 +1,216 @@
+- **Specify Instant Version:** Specify which release version of instant OpenHIE to use.
+- **Execute with current options:** Run the docker command to create and start up the instant OpenHIE docker container to install the rest of the services based on the custom options selected.
+- **View current options set:** Print out the current selected options to the CLI.
+- **Reset to default options:** Reset the current selected options to the default options to clear selection.
+
+#### Back & Quit:
+These options are available at every prompt.
+
+- **Back:** This option will take the user back one command so that they can make a different selection.
+- **Quit:** This option will terminate the execution of the GO CLI as well as stopping the instant OpenHIE docker container.
+
+## 3. Run the CLI in non-interactive mode
+
+### 3.1 Open your favourite terminal and run the executable file (linux and macOS)
+
+e.g. using an absolute path to the file:
+```sh
+/Projects/instant/goinstant-linux
+```
+or using a relative path to the file:
+```sh
+./instant/goinstant-linux
+```
+
+>Note: in Windows it's best to run non-interactive through Powershell.
+
+Currently the file downloaded from Github will not have the correct file permissions on linux and macOS, we are working on this, but in the meantime, you can execute the following command to update the file permissions to be executable:
+
+For Linux:
+```sh
+chmod 700 goinstant-linux
+```
+For MacOS (currently not supported):
+```sh
+chmod 700 goinstant-macos
+```
+
+### 3.2 Explanation of all flags and custom options
+
+#### Custom Install details:
+
+Format:
+```sh
+<binary-file> [args]
+```
+
+e.g.
+
+```sh
+.\goinstant.exe docker init core data-analytics only --env-file="C:\Projects\hiv-case-reporting\.env" -e="DOMAIN_NAME=testDomain" -e="OPENHIM_URL=testDomain.openhim.org" -c="C:\Projects\packages\data-analytics" -c="C:\Projects\packages\data-pipeline"
+```
+
+At least two of the following arguments is required (values in brackets are from above example):
+
+- **Target(docker):** docker is the only supported option at the moment.
+- **Deploy action(init):** init, destroy, up, down or test
+- **Deploy packages(core data-analytics):** Specify the package IDs that you want to operate on as part of the installation.
+- **Specify environment variable file location(--env-file="C:\Projects\hiv-case-reporting\.env"):** Specify a filepath to an environment variables file.
+- **Specify environment variables(-e="DOMAIN_NAME=testDomain"):** Specify an environment variable that should be aplied to the instant OpenHIE docker container.
+- **Specify custom package locations(-c="C:\Projects\packages\data-pipeline"):** Specify a comma delimited list of paths (filepaths or URLs) to custom packages.
+- **Toggle only flag(only):** This will enable isolation of the deploy action to only operate on specified packages and not take their dependency tree into consideration.
```

Comment:
```suggestion
- **Toggle only flag(only):** This will enable isolation of the deploy action to only operate on specified packages and not take their dependency tree into consideration.
- **Instant Image Version(--instant-version="latest"):** Specify the version of the instant openHIE image to use. 
```